tools/csv-gen/csv-gen.go

```diff
@@ -0,0 +1,690 @@
+		if err != nil {
+			fmt.Println(err)
+
+		}
+		if err = json.Unmarshal(result, &templateInterface); err != nil {
+			fmt.Println(err)
+		}
+	} else {
+		templateInterface = templateStruct
+	}
+
+	// find and replace images with SHAs where necessary
+	templateByte, err := json.Marshal(templateInterface)
+	if err != nil {
+		fmt.Println(err)
+	}
+	for from, to := range imageShaMap {
+		if to != "" {
+			templateByte = bytes.ReplaceAll(templateByte, []byte(from), []byte(to))
+		}
+	}
+	if err = json.Unmarshal(templateByte, &templateInterface); err != nil {
+		fmt.Println(err)
+	}
+	createFile(csvFile, &templateInterface)
+
+	packageFile := "deploy/olm-catalog/" + csv.CsvDir + "/" + csv.Name + ".package.yaml"
+	p, err := os.Create(packageFile)
+	defer p.Close()
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	pwr := bufio.NewWriter(p)
+	pwr.WriteString("#! package-manifest: " + csvFile + "\n")
+	packagedata := packageStruct{
+		PackageName: operatorName,
+		Channels: []channel{
+			{
+				Name:       "current-76",
+				CurrentCSV: operatorName + ".v" + version.PriorVersion,
+			},
+			{
+				Name:       maturity,
+				CurrentCSV: csvVersionedName,
+			},
+		},
+		DefaultChannel: maturity,
+	}
+	util.MarshallObject(packagedata, pwr)
+	pwr.Flush()
+}
+
+type csvSetting struct {
+	Name         string `json:"name"`
+	DisplayName  string `json:"displayName"`
+	OperatorName string `json:"operatorName"`
+	CsvDir       string `json:"csvDir"`
+	Registry     string `json:"repository"`
+	Context      string `json:"context"`
+	ImageName    string `json:"imageName"`
+	Tag          string `json:"tag"`
+}
+type csvPermissions struct {
+	ServiceAccountName string              `json:"serviceAccountName"`
+	Rules              []rbacv1.PolicyRule `json:"rules"`
+}
+type csvDeployments struct {
+	Name string                `json:"name"`
+	Spec appsv1.DeploymentSpec `json:"spec,omitempty"`
+}
+type csvStrategySpec struct {
+	Permissions        []csvPermissions                `json:"permissions"`
+	Deployments        []csvDeployments                `json:"deployments"`
+	ClusterPermissions []StrategyDeploymentPermissions `json:"clusterPermissions,omitempty"`
+}
+
+type StrategyDeploymentPermissions struct {
+	ServiceAccountName string            `json:"serviceAccountName"`
+	Rules              []rbac.PolicyRule `json:"rules"`
+}
+type channel struct {
+	Name       string `json:"name"`
+	CurrentCSV string `json:"currentCSV"`
+}
+type packageStruct struct {
+	PackageName    string    `json:"packageName"`
+	Channels       []channel `json:"channels"`
+	DefaultChannel string    `json:"defaultChannel"`
+}
+type image struct {
+	Name  string `json:"name"`
+	Image string `json:"image"`
+}
+
+func getRelatedImage(imageURL string) image {
+	imageName, _, _ := activemqartemis.GetImage(imageURL)
+	return image{
+		Name:  imageName,
+		Image: imageURL,
+	}
+}
+
+// fileExists checks if a file exists and is not a directory before we
+// try using it to prevent further errors.
```

Comment:
I feel like there should be some consistency in commenting functions - if this one needs a comment, so do the rest.